pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -0,0 +1,211 @@
+	r, err := newEgressRouterReconciler(mgr, status)
+	if err != nil {
+		return err
+	}
+
+	// Create a new controller
+	c, err := controller.New("egress-router-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource EgressRouter.network.operator.openshift.io/v1
+	err = c.Watch(&source.Kind{Type: &netopv1.EgressRouter{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &EgressRouterReconciler{}
+var manifestDir = "bindata/"
+
+type egressrouter struct {
+	spec netopv1.EgressRouterSpec
+}
+
+type EgressRouterReconciler struct {
+	mgr    manager.Manager
+	client client.Client
+	status *statusmanager.StatusManager
+
+	egressrouters    map[types.NamespacedName]*egressrouter
+	egressrouterErrs map[types.NamespacedName]error
+}
+
+var ResyncPeriod = 5 * time.Minute
+
+func newEgressRouterReconciler(mgr manager.Manager, status *statusmanager.StatusManager) (reconcile.Reconciler, error) {
+
+	return &EgressRouterReconciler{
+		mgr:    mgr,
+		status: status,
+		client: mgr.GetClient(),
+
+		egressrouters:    map[types.NamespacedName]*egressrouter{},
+		egressrouterErrs: map[types.NamespacedName]error{},
+	}, nil
+}
+
+func (r EgressRouterReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling egressrouter.network.operator.openshift.io %s\n", request.NamespacedName)
+
+	obj := &netopv1.EgressRouter{}
+	err := r.mgr.GetClient().Get(context.TODO(), request.NamespacedName, obj)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("Egress Router %s seems to have been deleted\n", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+
+	// Check to see if we already know this object
+	existing := r.egressrouters[request.NamespacedName]
+	if existing != nil {
+		if !reflect.DeepEqual(obj.Spec, existing.spec) {
+			log.Printf("Egress Router %s has changed, refreshing\n", request.NamespacedName)
+			delete(r.egressrouters, request.NamespacedName)
+			existing = nil
+		}
+	}
+
+	if existing == nil {
+		log.Printf("Creating a new Egress Router")
+		// Set owner reference to the controller
+		boolTrue := bool(true)
+		EgressRouterOwnerReferences := []v1.OwnerReference{
+			{
+				APIVersion: "network.operator.openshift.io/v1",
+				Kind:       "EgressRouter",
+				Name:       obj.Name,
+				UID:        obj.UID,
+				Controller: &boolTrue,
+			},
+		}
+		objs, err := ensureEgressRouter(manifestDir, request.Namespace, obj)
+		for _, obj := range objs {
+			log.Printf("Assigning owner references")
+			obj.SetOwnerReferences(EgressRouterOwnerReferences)
+			log.Printf("Applying manifest")
+			if err := apply.ApplyObject(context.TODO(), r.client, obj); err != nil {
+				log.Printf("could not apply operator config: %v", err)
```

Comment:
This line is wrong - "could not apply egress router object:"

Reply:
done